payjoin/src/send/error.rs

```diff
@@ -235,3 +236,220 @@ impl std::error::Error for CreateRequestError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::WellKnown(e, msg) => write!(
+                f,
+                r#"Well known error: {{ "errorCode": "{}",
+                "message": "{}" }}"#,
+                e, msg
+            ),
+            Self::Unrecognized(code, msg) => write!(
+                f,
+                r#"Unrecognized error: {{ "errorCode": "{}", "message": "{}" }}"#,
+                code, msg
+            ),
+            Self::Validation(e) => write!(f, "Validation({:?})", e),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum WellKnownError {
+    Unavailable,
+    NotEnoughMoney,
+    VersionUnsupported,
+    OriginalPsbtRejected,
+}
+
+impl FromStr for WellKnownError {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, ()> {
+        match s {
+            "unavailable" => Ok(WellKnownError::Unavailable),
+            "not-enough-money" => Ok(WellKnownError::NotEnoughMoney),
+            "version-unsupported" => Ok(WellKnownError::VersionUnsupported),
+            "original-psbt-rejected" => Ok(WellKnownError::OriginalPsbtRejected),
+            _ => Err(()),
+        }
+    }
+}
+
+impl Display for WellKnownError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Unavailable => write!(f, "The payjoin endpoint is not available for now."),
+            Self::NotEnoughMoney => write!(f, "The receiver added some inputs but could not bump the fee of the payjoin proposal."),
+            Self::VersionUnsupported => write!(f, "This version of payjoin is not supported."),
+            Self::OriginalPsbtRejected => write!(f, "The receiver rejected the original PSBT."),
+        }
+    }
+}
+
+impl From<RawResponse> for ResponseError {
+    fn from(value: RawResponse) -> Self {
+        let RawResponse { error_code, message } = value;
+        match WellKnownError::from_str(&error_code) {
+            Ok(e) => ResponseError::WellKnown(e, message),
+            Err(_) => ResponseError::Unrecognized(error_code, message),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use bitcoind::bitcoincore_rpc::jsonrpc::serde_json::json;
+
+    use super::*;
+
+    #[test]
+    fn test_parse_json() {
+        let known_json_error = json!({
+            "errorCode": "version-unsupported",
+            "message": "This version of payjoin is not supported."
+        });
+        match RawResponse::from_json(known_json_error.clone()) {
+            Ok(r) => match r.into() {
+                ResponseError::WellKnown(e, _) => assert_eq!(e, WellKnownError::VersionUnsupported),
+                _ => panic!("Error parsing json: {}", known_json_error),
+            },
+            Err(e) => panic!("Error parsing json: {}", e),
+        }
+        let known_str_error = "{\"errorCode\":\"version-unsupported\",
+                    \"message\":\"This version of payjoin is not supported.\"}";
```

Comment:
This is a good opportunity to take advantage of string literals

```suggestion
        let known_str_error = r#"{"errorCode":"version-unsupported", "message":"This version of payjoin is not supported."}"#;

```